client/tx/tx.go

```diff
@@ -294,3 +309,70 @@ type GasEstimateResponse struct {
 func (gr GasEstimateResponse) String() string {
 	return fmt.Sprintf("gas estimate: %d", gr.GasEstimate)
 }
+
+func makeAuxSignerData(clientCtx client.Context, f Factory, msgs ...sdk.Msg) (tx.AuxSignerData, error) {
+	b := NewAuxTxBuilder()
+	fromAddress, name, _, err := client.GetFromFields(clientCtx.Keyring, clientCtx.From, false)
+	if err != nil {
+		return tx.AuxSignerData{}, err
+	}
+	b.SetAddress(fromAddress.String())
+	if clientCtx.Offline {
+		b.SetAccountNumber(f.accountNumber)
+		b.SetSequence(f.sequence)
+	} else {
+		if err != nil {
+			return tx.AuxSignerData{}, err
+		}
+
+		accNum, seq, err := clientCtx.AccountRetriever.GetAccountNumberSequence(clientCtx, fromAddress)
+		if err != nil {
+			return tx.AuxSignerData{}, err
+		}
+		b.SetAccountNumber(accNum)
+		b.SetSequence(seq)
+	}
+
+	err = b.SetMsgs(msgs...)
+	if err != nil {
+		return tx.AuxSignerData{}, err
+	}
+
+	if f.tip != nil && f.tip.String() != "" {
+		b.SetTip(&tx.Tip{Amount: f.tip.Amount, Tipper: fromAddress.String()})
```

Comment:
Actually, we need to have a `--tipper flag` too.

The tipper and fee payer are not on the same chain. If the tipper uses fromAddress, then the bech32 will be the tipper's chain's prefix. We need the tipper to use the target (fee payer's) chain prefix.